src/main/java/dev/galacticraft/mod/block/special/walkway/PipeWalkway.java

```diff
@@ -76,92 +71,123 @@ private static int getFacingMask(Direction dir) {
+                    if (pipe.canConnect(direction.getOpposite())) {
+                        blockEntity.getConnections()[direction.ordinal()] = true;
+                        continue;
+                    }
+                } else if (FluidUtil.canAccessFluid(world, pos.offset(direction), direction)) {
+                    blockEntity.getConnections()[direction.ordinal()] = true;
+                    continue;
+                }
+            }
+            blockEntity.getConnections()[direction.ordinal()] = false;
+        }
+        world.updateNeighborsAlways(pos, state.getBlock());
     }
 
     @Override
-    public BlockState mirror(BlockState state, BlockMirror mirror) {
-        return ConnectingBlockUtil.mirror(state, mirror);
-    }
-
-    public boolean canConnect(BlockState state, BlockState neighborState, BlockPos pos, BlockPos neighborPos, WorldAccess world, Direction facing) {
-        /* This code tests if the connecting block is on the top of the walkway (it used to be unable to connect there)
-        try {
-            if (pos.offset(state.get(FACING)).equals(neighborPos))
-                return false;
-            if (neighborPos.offset(neighborState.get(FACING)).equals(pos))
-                return false;
-        } catch (IllegalArgumentException ignored) {}
-        */
-        if (FluidUtil.isExtractableOrInsertable((World) world, neighborPos, facing.getOpposite()))
-            return true;
-        return (neighborState.getBlock() instanceof FluidPipe); //&& (neighborState.get(COLOR) == state.get(COLOR));
+    public BlockState getStateForNeighborUpdate(BlockState state, Direction facing, BlockState neighborState, WorldAccess world, BlockPos pos, BlockPos neighborPos) {
+        if (!this.isEmpty(state)) {
+            world.getFluidTickScheduler().schedule(pos, Registry.FLUID.get(state.get(FLUID)), Registry.FLUID.get(state.get(FLUID)).getTickRate(world));
+        }
+        return state;
     }
 
     @Override
-    public ActionResult onUse(BlockState state, World world, BlockPos pos, PlayerEntity player, Hand hand, BlockHitResult hit) {
-        if (!player.getStackInHand(hand).isEmpty()) {
-            if (player.getStackInHand(hand).getItem() instanceof DyeItem dye) {
-                ItemStack stack = player.getStackInHand(hand).copy();
-                DyeColor color = dye.getColor();
-                if (color != state.get(COLOR)) {
-                    stack.decrement(1);
-                    player.setStackInHand(hand, stack);
-                    world.setBlockState(pos, state.with(COLOR, color));
-                    return ActionResult.SUCCESS;
-                } else {
-                    return ActionResult.FAIL;
+    public void neighborUpdate(BlockState state, World world, BlockPos pos, Block block, BlockPos fromPos, boolean notify) {
+        super.neighborUpdate(state, world, pos, block, fromPos, notify);
+        if (fromPos.isWithinDistance(pos, 1.0000000000001)) {
```

Comment:
Well, it was to make sure that the adjacent position was actually directly adjacent, otherwise the `Direction#fromVector` calls would fail. I've changed it into a more sane absolute subtract `== 1` in 55a80888efb24d73370216acaaac60aba3712f2b.